test/rendering-test/test-rendering.js

```diff
@@ -211,6 +227,7 @@ function reportResult(name, percentage) {
 
   const paragraph = document.createElement('p');
   const testResult = document.createTextNode(outputString);
+  paragraph.style.color = passed ? '#74ff69' : '#ff2857';
```

Comment:
* Separate out the logging into separate function.

Also, but not for this diff:
* This method should not do any logging but just return a `Promise` with the results.
* This test code should not use React, but just use our Pure JS API.